crypto/src/mls/conversation/proposal.rs

```diff
@@ -17,6 +17,78 @@ use crate::{
     MlsError, RecursiveError,
 };
 
+/// Sending proposals
+impl CentralContext {
+    pub(crate) async fn send_and_merge_or_discard_proposal(
+        &self,
+        conversation: &mut MlsConversation,
+        proposal: MlsMessageOut,
+    ) -> Result<()> {
+        let guard = self
+            .mls_transport()
+            .await
+            .map_err(RecursiveError::root("getting mls transport"))?;
+        let transport = guard.as_ref().ok_or::<Error>(
+            RecursiveError::root("getting mls transport")(crate::Error::MlsTransportNotProvided).into(),
+        )?;
+        match transport
+            .send_message(
+                proposal
+                    .to_bytes()
+                    .map_err(MlsError::wrap("constructing byte vector of proposal"))?,
+            )
+            .await
+            .map_err(RecursiveError::root("sending mls message"))?
+        {
+            MlsTransportResponse::Success => Ok(()),
+            MlsTransportResponse::Abort { reason } => {
+                self.clear_all_pending_proposals(conversation).await?;
+                Err(Error::MessageRejected { reason })
+            }
+            MlsTransportResponse::Retry => self.commit_pending_proposals(conversation.id()).await,
```

Comment:
Why do we commit pending proposals when we get a `Retry`? Shouldn't that be done on `Success`?
I note the Jira item also has this in the pseudocode.